### ISSUE TITLE
Restriction on multiple spread fields initialization for maps

### ISSUE THREAD

**Issue by ruvi-d** [ballerina-platform/ballerina-spec#826]
**Description:**

The following code snippet was valid pre SLP2
```Java
public function main() {
    map<int> boys = {Jake: 80, Steve: 90};
    map<int> girls = {Amy: 85, Sam: 88};
    map<int> allMarks = {...boys, ...girls};
}
```
But with SLA4, the compiler throw the following error:
`invalid usage of mapping constructor expression: multiple spread fields of inclusive mapping types are not allowed`

Is there a reason why this is no longer valid? The restriction seems unnecessary. 

**Suggested Labels:**

   
**Code sample that shows issue:**

**Related Issues:**

**Comment by ruvi-d** [comment]
CC @manuranga

**Comment by jclark** [comment]
Yes: general rule is that it is compile time guaranteed that you cannot have duplicate keys in a mapping constructor (except with computed keys).
